src/main/java/ui/components/NavigableListView.java

```diff
@@ -188,21 +188,12 @@ public void setOnItemSelected(IntConsumer callback) {
         onItemSelected = callback;
     }
 
-    public void selectFirstItem() {
+    public void selectnthItem(int n) {
         requestFocus();
-        if (getItems().size() == 0) return;
-        getSelectionModel().clearAndSelect(0);
-        scrollAndShow(0);
-        selectedIndex = Optional.of(0);
-        onItemSelected.accept(selectedIndex.get());
-    }
-
-    public void selectLastItem() {
-        requestFocus();
-        if (getItems().size() == 0) return;
-        getSelectionModel().clearAndSelect(getItems().size() - 1);
-        scrollAndShow(getItems().size() - 1);
-        selectedIndex = Optional.of(getItems().size() - 1);
+        if (getItems().size() < n) return;
```

Comment:
This can be changed into an assertion, since off-by-one errors here are programmer errors. It should also assert the lower bound (`> 0`).